Spring-Boot/Spring-Boot-Submission/src/test/java/tw/waterball/judgegirl/springboot/submission/controllers/AbstractSubmissionControllerTest.java

```diff
@@ -344,16 +349,22 @@ protected ResultActions submitCode(int studentId, Token token, MockMultipartFile
                 multipartRequestWithSubmittedCodes(studentId, files)));
     }
 
+    @SneakyThrows
     protected MockHttpServletRequestBuilder multipartRequestWithSubmittedCodes(int studentId, MockMultipartFile... files) {
         var call = multipart(API_PREFIX, problem.getId(), studentId);
+        List<Part> parts = new ArrayList<>();
+
         for (MockMultipartFile file : files) {
             call = call.file(file);
         }
-        MockHttpServletRequestBuilder addingHeaders = call;
-        for (var entry : submissionBag.entrySet()) {
-            addingHeaders = call.header(HEADER_BAG_KEY_PREFIX + entry.getKey(), entry.getValue());
-        }
-        return addingHeaders;
+
+        var bagPart = new MockPart(SUBMISSION_BAG_MULTIPART_KEY_NAME, SUBMISSION_BAG_MULTIPART_KEY_NAME,
+                toJson(submissionBag).getBytes(UTF_8));
+        bagPart.getHeaders().setContentType(MediaType.APPLICATION_JSON);
+        parts.add(bagPart);
+        call.part(parts.toArray(new Part[0]));
```

Comment:
delete List<Part>, and change to user code below
```Java
call.part( new Part[]{bagPart});
```

Reply:
`call.part(bagPart);`